src/main/java/teammates/storage/entity/Comment.java

```diff
@@ -98,6 +106,8 @@ public Comment(String courseId, String giverEmail, CommentRecipientType recipien
         this.showRecipientNameTo = showRecipientNameTo;
         this.createdAt = date;
         this.commentText = comment;
+        this.lastEditorEmail = (lastEditorEmail == null) ? giverEmail : lastEditorEmail;
+        this.lastEditedAt = (lastEditedAt == null) ? date : lastEditedAt;
```

Comment:
So u're replacing null at entity level, instead of attribute value, which wont resolve NPE's problem